src/model/modifier/NestedTextEditorUtil.js

```diff
@@ -0,0 +1,589 @@
+                type: 'unstyled',
+                depth: 0,
+                text: currentBlock.getText(),
+                characterList: currentBlock.getCharacterList()
+              })
+            ];
+          }
+        }
+        return block;
+      })
+      .reduce((a, b) => a.concat(b), [])
+    );
+
+    return {
+      editorState: EditorState.push(
+        editorState,
+        newContentState.merge({
+          selectionBefore: selectionState,
+          selectionAfter: selectionState.merge({
+            anchorKey: isCousinType ? key : targetKey,
+            anchorOffset: selectionState.getAnchorOffset(),
+            focusKey: isCousinType ? key : targetKey,
+            focusOffset: selectionState.getFocusOffset(),
+            isBackward: false,
+          })
+        }),
+        'change-block-type'
+      ),
+      blockType
+    };
+  },
+
+  handleKeyCommand: function(
+    editorState: EditorState,
+    command: DraftEditorCommand,
+    blockRenderMap: DraftBlockRenderMap = DefaultBlockRenderMap
+  ): ? EditorState {
+    const selectionState = editorState.getSelection();
+    const contentState = editorState.getCurrentContent();
+    const key = selectionState.getAnchorKey();
+
+    const currentBlock = contentState.getBlockForKey(key);
+    const nestedBlocks = contentState.getBlockChildren(key);
+
+    const parentKey = currentBlock.getParentKey();
+    const parentBlock = contentState.getBlockForKey(parentKey);
+    const nextBlock = contentState.getBlockAfter(key);
+
+    // Option of rendering for the current block
+    const renderOpt = blockRenderMap.get(currentBlock.getType());
+    const parentRenderOpt = parentBlock && blockRenderMap.get(parentBlock.getType());
+
+    const hasNestingEnabled = renderOpt && renderOpt.nestingEnabled;
+    const hasWrapper = renderOpt && renderOpt.wrapper;
+
+    const parentHasWrapper = parentRenderOpt && parentRenderOpt.wrapper;
+
+    // Press enter
+    if (command === 'split-block') {
+      if (
+        currentBlock.hasParent() &&
+        (!hasNestingEnabled ||
+          currentBlock.getLength() === 0
+        ) &&
+        (!nextBlock ||
+          (
+            hasWrapper &&
+            nextBlock.getType() !== currentBlock.getType()
+          ) ||
+          (
+            nextBlock.getParentKey() !== currentBlock.getParentKey() &&
+            (currentBlock.getLength() === 0 || parentHasWrapper)
+          )
+        )
+      ) {
+        command = 'split-parent-block';
+      }
+
+      // In a block that already have some nested blocks
+      if (command === 'split-block' && nestedBlocks.size > 0) {
+        command = 'split-nested-block';
+      }
+    }
+
+    // Prevent creation of nested blocks
+    if (!hasNestingEnabled && command === 'split-nested-block') {
+      command = 'split-block';
+    }
+
+    switch (command) {
+      case 'backspace':
+        return NestedTextEditorUtil.onBackspace(editorState, blockRenderMap);
```

Comment:
If `NestedTextEditorUtil` is to support being extended, shouldn't these commands be `this.onBackspace`, `this.onDelete`, etc?